### ISSUE TITLE
It would be nice with a cicd build of overtone to clojars

### ISSUE THREAD

**Issue by jave** [overtone/overtone#572]
When wanting to try out unreleased stuff it would be nice to have a github cicd that pushes to clojars.

Especially as so much exiting stuff is happening in overtone land recently!

**Comment by plexus** [comment]
Use `:local/root`?

**Comment by pfeodrippe** [comment]
@plexus I've advised Joakim about :git/url on slack and asked him to create this issue, this will work only with deps.edn (the lein plugin for git is unreliable). :local/root would be the same.

Just wondering if we don’t want to publish master releases for every commit or if it’s just lacking someone to do it.

Let me know o/

**Comment by plexus** [comment]
Ok, there are a few different things at play here. My main attitude is that Clojure CLI was created in part to fix this exact issue. It's been around for something like 7 years, people should upgrade/switch. Any time we spend supporting outdated tooling is time we could have spent on making Overtone better (or making cool music), including this conversation.

I'm generally not a fan of doing releases from CI. I am a fan of frequent, small, automated releases. Overtone uses the tooling we also use for Lambda Island. It's a single command to release to clojars, tag the release, update GH releases, add a release heading to CHANGELOG.md, add a message to any merged PRs that they have been released in that version, etc. It takes a handful of seconds to run.

For LI stuff I generally do a release after every merge. The main reason I haven't done that here recently is because of the CHANGELOG. Not every PR includes a CHANGELOG entry, and so I need to manually review what has gone in and update the CHANGELOG before doing a release. The biggest thing someone could do to get more frequent releases is to create whatever github magic files are needed to get a warning/reminder when creating a PR to include a CHANGELOG entry.

I also think we should continue to raise the bus factor. As far as I'm concerned anyone who's made a genuine contribution should be allowed to merge to main and to release to clojars, but I don't have that kind of admin rights to grant people that, we'll have to ask @samaaron on a case by case basis.

That all said I think for a project like Overtone there's also a benefit to batching up changes and releasing them in chunks, especially if we then also make the effort of writing release notes and pushing them to the relevant channels. It's a great way to create interest in the project. For that you want releases that are regular but not too often. At least a few times a year, at most once a month, so people don't get fatigued by small release announcements that don't contain substance.

Now this is all my personal opinion, I don't speak for "the overtone project" as such. No one does right now, we're just a loose coalition of people trying to move this thing forward. So if people want to see releases from CI, and want to do the work to make it happen, then I'm not going to object that. In that case I would prefer them to be -SNAPSHOT releases so we keep "real" releases as something that has a human's stamp of approval.

**Comment by pfeodrippe** [comment]
Yes, no need for a “real” release, I agree that these, if done for every commit, should be dev/snapshots releases, it would work fine o/

Let’s see if Sam can give us direction here, but I agree it’s not a priority for sure.

Thanks for the response, man, and thanks for taking OV back to life ;)

**Comment by plexus** [comment]
FYI all you need for a release is to run `bin/proj release`, the main challenge for CI will be to make sure it has the right keys/tokens for clojars, push access to github, and github API access (for making the release + commenting on merged PRs).

**Comment by jave** [comment]
btw im fine with using a git hash locally in deps.edn, so no biggie. tnx for keeping overtone alive!
